`path must be a string` when requiring uninstall dependency

## Description

When using `babel-register` as required dependency of AVA, I got this error.

```
assert.js:93
  throw new assert.AssertionError({
        ^
AssertionError: path must be a string
    at Module.require (module.js:362:3)
    at require (module.js:380:17)
    at Array.forEach (native)
    at Object.<anonymous> (/Users/sam/Projects/test/node_modules/ava/lib/test-worker.js:48:22)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:929:3
```

It took me a while to figure out that it was because I didn't install the `babel-register` dependency and [resolve-cwd](https://www.npmjs.com/package/resolve-cwd) returns `null` for dependencies not found.

Because of [this](https://github.com/sindresorhus/ava/blob/master/api.js#L31) line, the `require` argument results in `[null]` and this fails when trying to `require` `null` off course. It would be nice if it failed with a decent error like `Could not find dependency 'babel-register'` or whatever dependency could not be found.

## Environment

This was only the case on `v0.10.36` for me. Not sure why though but probably because it uses pre-npm@3 and thus not flattens the dependency tree. I don't really think this issue is platform dependent.

```
Node.js v0.10.36
darwin 15.3.0
```

--- a/api.js
+++ b/api.js
@@ -28,8 +28,15 @@ function Api(options) {
 	EventEmitter.call(this);
 
 	this.options = options || {};
-	this.options.require = (this.options.require || []).map(resolveCwd);
 	this.options.match = this.options.match || [];
+	this.options.require = (this.options.require || []).map(function (moduleId) {
+		var ret = resolveCwd(moduleId);
+		if (ret === null) {
+			throw new Error('Could not resolve required module \'' + moduleId + '\'');
+		}
+
+		return ret;
+	});
 
 	this.excludePatterns = [
 		'!**/node_modules/**',

--- a/test/api.js
+++ b/test/api.js
@@ -533,6 +533,15 @@ test('Node.js-style --require CLI argument', function (t) {
 		});
 });
 
+test('Node.js-style --require CLI argument module not found', function (t) {
+	t.plan(1);
+
+	t.throws(function () {
+		/* eslint no-new: 0 */
+		new Api({require: ['foo-bar']});
+	}, /^Could not resolve required module 'foo-bar'$/);
+});
+
 test('power-assert support', function (t) {
 	t.plan(3);
 

